src/OmniSharp.Cake/Services/RequestHandlers/Diagnostics/CodeCheckHandler.cs

```diff
@@ -18,5 +22,33 @@ public class CodeCheckHandler : CakeRequestHandler<CodeCheckRequest, QuickFixRes
 
         public override bool IsValid(CodeCheckRequest request) =>
             !string.IsNullOrEmpty(request.FileName);
+
+        protected override Task<QuickFixResponse> TranslateResponse(QuickFixResponse response, CodeCheckRequest request)
+        {
+            if (response?.QuickFixes == null)
+            {
+                return Task.FromResult(response);
+            }
+
+            var quickFixes = response.QuickFixes.Where(x => PathsAreEqual(x.FileName, request.FileName));
+            response.QuickFixes = quickFixes;
+            return Task.FromResult(response);
+
+            bool PathsAreEqual(string x, string y)
+            {
+                if (x == null && y == null)
```

Comment:
Could be `ReferenceEquals(x, y)` to be even more efficient in case the same string instance happens to be reused.